docs/tutorials/send-payjoin-with-pdk.md

```diff
@@ -0,0 +1,359 @@
+
+Payjoin requires messages to be sent between sender and receiver. The
+receiver hosts an HTTP endpoint to receive these messages. To make a
+request, we need the receiver payjoin Uri. 
+
+The Uri contains the amount the receiver is requesting and the address
+they wish to receive to. Moreover, the Uri also includes the endpoint
+we will make the request to, and that is extracted in the
+`RequestContext::from_psbt_and_uri`.
+
+Notice that other parameters like `pjos` can also be extracted.
+
+```rust
+// mod sender {
+    // ..
+    pub fn try_payjoin(pj_uri: payjoin::Uri<'static, NetworkChecked>) -> Txid {
+        let sender_wallet = get_sender_wallet();
+        let amount_to_send = pj_uri.amount.unwrap();
+        let receiver_address = pj_uri.address.clone();
+    }
+// }
+```
+
+##### 3.4 Construct an "Original PSBT" paying `.amount` to `.address`
+as specified in the URI.
+
+First we add the receiver address to the outputs list along side the
+request amount. Then we create a funded PSBT with options set to
+specify `fee_rate` and other parameters.
+
+```rust
+// ...
+// pub fn try_payjoin(pj_uri: payjoin::Uri<'static, NetworkChecked>) -> Txid {
+    // ...
+    let mut outputs = HashMap::with_capacity(1);
+    outputs.insert(receiver_address.to_string(), amount_to_send);
+    let options = bitcoincore_rpc::json::WalletCreateFundedPsbtOptions {
+                lock_unspent: Some(false),
+                fee_rate: Some(bitcoincore_rpc::bitcoin::Amount::from_sat(10000)),
+                ..Default::default()
+    };
+    let sender_psbt = sender_wallet
+        .wallet_create_funded_psbt(
+                &[], // inputs
+                &outputs,
+                None, // locktime
+                Some(options),
+                None,
+                )
+        .unwrap();
+    let psbt = sender_wallet
+        .wallet_process_psbt(&sender_psbt.psbt, None, None, None)
+        .unwrap()
+        .psbt;
+    let psbt = Psbt::from_str(&psbt).unwrap();
+// }
+```
+
+##### 3.5 Construct the HTTPS request
+
+Prepare the request we want to send and also get the context back so we can
+handle the response when its received.
+
+```rust
+// ...
+// pub fn try_payjoin(pj_uri: payjoin::Uri<'static, NetworkChecked>) -> Txid {
+    // ...
+        let (req, ctx) = RequestBuilder::from_psbt_and_uri(psbt, pj_uri)
+        .unwrap()
+        .build_with_additional_fee(
+                bitcoincore_rpc::bitcoin::Amount::from_sat(1),
+                None,
+                bitcoincore_rpc::bitcoin::FeeRate::MIN,
+                true,
+                )
+        .unwrap()
+        .extract_v1()
+        .unwrap();
+// }
+```
+
+##### 3.6 Send the request 
+
+```rust
+// ...
+// pub fn try_payjoin(pj_uri: payjoin::Uri<'static, NetworkChecked>) -> Txid {
+    // ...
+        let res = https_agent()
+        .post(req.url.as_str()) // pj_uri.endpoint
+        .header("Content-Type", "text/plain")
+        .body(String::from_utf8(req.body).unwrap())
```

Comment:
```suggestion
        .post(req.url) // pj_uri.endpoint
        .header("Content-Type", "text/plain")
        .body(req.body)
```
The dev kit intends to reduce the need for any HTTP I/O adapter. It should give you appropriate types to pass directly to an HTTP client. If not that might be a bug we can fix.

Reply:
I accepted this change without testing it. and then I saw your comment. why are you suggesting a code change, with the code you want to see without testing it first?

I spent sometime testing each step and made sure its working. I do expect, if you suggest code changes, to at least validate them locally first?